analysis.ml

```diff
@@ -26,24 +26,12 @@ let predecessors program =
   done;
   preds
 
-module VarSet = Set.Make(Variable)
-
-(* Perform forward analysis on some code
- *
- * init_state : Initial input state and first instruction
- * merge      : current state -> input state -> merge state if changed
- * update     : abstract instruction -> input state -> output state
- * program    : array of abstract instructions
- * prog_at    : program -> index -> instruction at index
- *
- * Returns an array of states for every instruction of the program.
- * Bottom is represented as None *)
-
-let forward_analysis (init_state : ('a * pc))
-                     (merge : 'a -> 'a -> 'a option)
-                     (update : pc -> 'a -> 'a)
-                     (program : program)
-                     : 'a option array =
+let do_analysis (successors : pc -> pc list)
```

Comment:
I think "successors" is confusing here. What about simply `neighbors` or `next`? Also, `do_analysis` is a meh name, what about `analyze`?